src/Layer/Launchpad/Launchpad.lua

```diff
@@ -39,9 +39,9 @@ end
 
 --- connect launchpad to a midi device
 --
-function Launchpad:connect(midi_device_name)
-    self.midi_out    = renoise.Midi.create_output_device(midi_device_name)
-    self.midi_input  = renoise.Midi.create_input_device(midi_device_name, self.__midi_input_callback)
+function Launchpad:connect(midi_in_device_name, midi_out_device_name)
```

Comment:
what do you need the out_device here? why should the render device and the input device differ?